src/projects/dualBackedYieldToOne/MDualBackedYieldToOne.sol

```diff
@@ -0,0 +1,166 @@
+        0xcf2d6e9d1351d8d0083357b686a68d8501c3acb71aa4753ff0da62fd455c8900;
+
+    function _getMDualBackedYieldToOneStorageLocation()
+        internal
+        pure
+        returns (MDualBackedYieldToOneStorageStruct storage $)
+    {
+        assembly {
+            $.slot := _M_DUAL_BACKED_YIELD_TO_ONE_STORAGE_LOCATION
+        }
+    }
+}
+
+/// @custom:oz-upgrades-unsafe-allow missing-initializer
+/// @custom:oz-upgrades-unsafe-allow constructor
+
+contract MDualBackedYieldToOne is IMDualBackedYieldToOne, MDualBackedToYieldOneStorageLayout, MYieldToOne {
+    bytes32 public constant COLLATERAL_MANAGER_ROLE = keccak256("COLLATERAL_MANAGER_ROLE");
+
+    /**
+     * @custom:oz-upgrades-unsafe-allow constructor
+     * @notice Constructs MYieldToOne Implementation contract
+     * @dev    Sets immutable storage.
+     * @param  mToken       The address of $M token.
+     * @param  swapFacility The address of Swap Facility.
+     */
+    constructor(address mToken, address swapFacility) MYieldToOne(mToken, swapFacility) {}
+
+    function initialize(
+        string memory name,
+        string memory symbol,
+        address yieldRecipient,
+        address admin,
+        address freezeManager,
+        address yieldRecipientManager,
+        address collateralManager,
+        address secondaryBacker
+    ) public virtual initializer {
+        __MDualBackedYieldToOne_init(
+            name,
+            symbol,
+            yieldRecipient,
+            admin,
+            freezeManager,
+            yieldRecipientManager,
+            collateralManager,
+            secondaryBacker
+        );
+    }
+
+    function __MDualBackedYieldToOne_init(
+        string memory name,
+        string memory symbol,
+        address yieldRecipient,
+        address admin,
+        address freezeManager,
+        address yieldRecipientManager,
+        address collateralManager,
+        address secondaryBacker
+    ) internal onlyInitializing {
+        if (collateralManager == address(0)) revert ZeroCollateralManager();
+        if (secondaryBacker == address(0)) revert ZeroSecondaryBacker();
+
+        __MYieldToOne_init(name, symbol, yieldRecipient, admin, freezeManager, yieldRecipientManager);
+
+        _grantRole(COLLATERAL_MANAGER_ROLE, collateralManager);
+
+        MDualBackedYieldToOneStorageStruct storage $ = _getMDualBackedYieldToOneStorageLocation();
+
+        $.secondaryBacker = secondaryBacker;
+    }
+
+    function secondaryBacker() public view returns (address) {
+        return _getMDualBackedYieldToOneStorageLocation().secondaryBacker;
+    }
+
+    function secondarySupply() public view returns (uint256) {
+        return _getMDualBackedYieldToOneStorageLocation().secondarySupply;
+    }
+
+    function yield() public view override(IMYieldToOne, MYieldToOne) returns (uint256) {
+        unchecked {
+            uint256 mBalance_ = _mBalanceOf(address(this));
+            uint256 mBacking_ = totalSupply() - secondarySupply();
```

Comment:
mBacking can have its own internal method

Reply:
Was added in a previous commit.